stat_280_project_netflix_recsys.Rmd

```diff
@@ -352,5 +347,91 @@ simulations.dt %>%
 ```
 
 
+## Netflix Data
+
+In this section, we apply the BPMF Gibbs Sampler to our sampled ratings data set on 15 users and 25 movies.
+
+```{r}
+# Sample ratings
+sampled_ratings <- sample_ratings(ratings, 25, 25)
+sampled_ratings %>% summarise(n_users=n_distinct(userId),
+                              n_movies=n_distinct(movieId),
+                              n_ratings=n())
+```
+
 
+```{r}
+# Apply mapping from userId -> userKey and movieId -> movieKey
+# This allows it to work with the Gibbs Sampler which assumes 1:n_users as
+# indices for users and 1:n_movies as indices for movies
+sampled_ratings_user_key <- sampled_ratings %>%
+  select(userId) %>%
+  distinct %>%
+  mutate(user_key=row_number())
+
+sampled_ratings_movie_key <- sampled_ratings %>%
+  select(movieId) %>%
+  distinct %>%
+  mutate(movie_key=row_number())
+
+sampled_ratings_mapped <- sampled_ratings %>%
+  left_join(sampled_ratings_user_key, by="userId") %>% 
+  left_join(sampled_ratings_movie_key, by="movieId") %>% 
+  select(user_key, movie_key, rating)
+
+sampled_ratings_mapped %>% head
+```
+
+```{r}
+sampled_ratings_mapped %>%
+  vis_matrix(data = .,
+             row_col = "user_key",
+             column_col = "movie_key",
+             value_col = "rating")
+```
+
+```{r}
+# Implement Gibbs sampler
+n_replications <- 800
+
+# Define initial parameters
+k_estimate <- 10
+alpha <- 1
+mu_0 <- rep(0, k_estimate)
+beta_0 <- 1
+nu_0 <- 1
+W_0 <- diag(k_estimate)
+
+sampled_ratings_matrix <- sampled_ratings_mapped %>%
+  mutate(rating=(rating - 1)/5) %>% 
```

Comment:
Let's use the mutate(ratings = sapply(ratings, transform_rating_to_score)) function

Reply:
![image](https://user-images.githubusercontent.com/57904108/69489314-dd6a8500-0eb0-11ea-9115-dbe9306d7dda.png)
Doesn't seem correct

Reply:
![image](https://user-images.githubusercontent.com/57904108/69489387-7057ef00-0eb2-11ea-8cc7-ae9bc240bb57.png)
Looks good